pkg/objstore/s3/s3.go

```diff
@@ -161,36 +161,57 @@ func NewBucket(logger log.Logger, conf []byte, component string) (*Bucket, error
 	return NewBucketWithConfig(logger, config, component)
 }
 
+type overrideSignerType struct {
+	signerType credentials.SignatureType
+	upstream   credentials.Provider
+}
+
+func (s *overrideSignerType) Retrieve() (credentials.Value, error) {
+	v, err := s.upstream.Retrieve()
+	if err != nil {
+		return v, err
+	}
+	if !v.SignerType.IsAnonymous() {
+		v.SignerType = s.signerType
+	}
+	return v, nil
+}
+
+func (s *overrideSignerType) IsExpired() bool {
```

Comment:
No need for this if we would embed `wrapped`/`upstream`

Reply:
Very good point, just done that as part of the last push